relayer/src/foreign_client.rs

```diff
@@ -787,17 +787,17 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
             Ok(res) => {
                 // If elapsed < refresh_window for the client, `try_refresh()` will
                 // be successful with an empty vector.
-                if let Some(ibc_events) = res.clone() {
+                if let Some(ibc_events) = &res {
                     // The assumption is that only one IbcEventType::ChainError will be
                     // in the resulting Vec<IbcEvent> if an error occurred.
                     match ibc_events
-                        .into_iter()
-                        .find(|e| e.event_type() == IbcEventType::ChainError)
+                        .iter()
+                        .find(|&e| e.event_type() == IbcEventType::ChainError)
                     {
                         Some(ev) => {
                             return Err(ForeignClientError::chain_error_event(
                                 self.dst_chain().id(),
-                                ev,
+                                ev.to_owned(),
```

Comment:
And this could be just moved without a copy.

```suggestion
                                ev,
```

But I may be missing some remaining borrow checker issue here, try it out and see if something pops out. Weirdly I get some error on not being able to use `==` with `IbcEventType` values.